packages/teams-js/src/internal/communication.ts

```diff
@@ -676,7 +676,11 @@ function flushMessageQueue(targetWindow: Window | any): void {
  * Limited to Microsoft-internal use
  */
 export function waitForMessageQueue(targetWindow: Window, callback: () => void): void {
-  const messageQueueMonitor = Communication.currentWindow.setInterval(() => {
+  let messageQueueMonitor: ReturnType<typeof setInterval>;
+  /* const cannot be used to declare messageQueueMonitor here because of the JS temporal dead zone. In order for messageQueueMonitor to be referenced inside setInterval,
+     it has to be defined before the setInterval call. */
```

Comment:
This is a behavior difference introduced in ES6.

In ES5, the `var` keyword is used for variable declaration, which is hoisted to the top of its scope. This means even if a variable is declared after it's used, the JavaScript engine moves the declaration to the top, so it doesn't throw an error.

However, in ES6, the `let` and `const` keywords were introduced for variable declaration. These are block-scoped and are not hoisted like `var`. More importantly, they have a temporal dead zone (TDZ), a period between entering scope and being declared where they cannot be accessed. This was causing the "Cannot access 'messageQueueMonitor' before initialization" errors I was seeing in the unit tests.

When we were compiling to ES5, TypeScript was transpiling `const` and `let` to `var`, which doesn't have a TDZ, so the error didn't occur. But when I switched to ES6, `const` and `let` are kept as is, and thus the TDZ is in effect, causing the error.